src/wallet/wallet.cpp

```diff
@@ -2639,7 +2639,10 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
     // enough, that fee sniping isn't a problem yet, but by implementing a fix
     // now we ensure code won't be written that makes assumptions about
     // nLockTime that preclude a fix later.
-    txNew.nLockTime = chainActive.Height();
+    {
+        LOCK(cs_main);
```

Comment:
Remove, all callers have `cs_main` locked, assert lock is held instead. Ping @ryanofsky as he says the goal is to removed recursive locks.

Reply:
@promag Thanks! Now fixed.

Reply:
@promag I'm getting an assertion error when running with `test/functional/test_runner.py` with your suggested patch applied. Could you investigate? Reverting to previous version for now.

Reply:
I'll check.

Reply:
@promag Any idea on how to tackle the assertion failure? :-)

Reply:
@promag See @TheBlueMatt's comment about `cs_main` locking in `CWallet::CreateTransaction(…)`: https://github.com/bitcoin/bitcoin/pull/11596#discussion_r149189367